### PR TITLE
Cortx 33963 update motr lib

### PR DESCRIPTION
**Pull request by zohebkhann**
# Problem Statement
- Updating the Motr k8 lib and tests as per the new service framework(v0.9.0 onwards)

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
[_TestMotrLib_20220818225204.log](https://github.com/Seagate/cortx-test/files/9424209/_TestMotrLib_20220818225204.log)
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)
[logs.zip](https://github.com/Seagate/cortx-test/files/9423468/logs.zip)
[collection.log](https://github.com/Seagate/cortx-test/files/9432309/collection.log)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: commons/constants.py [hunk 1]
```diff
@@ -49,6 +49,7 @@
 POD_NAME_PREFIX = "cortx-data"
 CORTX_DATA_NODE_PREFIX = "cortx-data-headless-svc-"
 SERVER_POD_NAME_PREFIX = "cortx-server"
+CORTX_CLIENT_SVC_POSTFIX = "cortx-client-headless.cortx.svc.cluster.local"
```
Comment (gaurkrgaurav) [inline commons/constants.py:52]: Can we change it to more general similar to another name?

Comment (gaurkrgaurav) [reply commons/constants.py:52]: Also please add test run log

Comment (zohebkhann) [reply commons/constants.py:52]: This is common to every motr client service name also we needed this info to find out the corresponding primary's node client name

Comment (zohebkhann) [reply commons/constants.py:52]: > Also please add test run log

Added logs.

**Review by sonalk0209 — APPROVED** [review]

**Review by gaurkrgaurav — APPROVED** [review]

File: comptests/motr/test_motr_k8s_sanity.py [hunk 2]
```diff
@@ -82,8 +82,8 @@ def test_m0crate_utility(self, param_loop):
         This is to run the m0crate utility tests.
         param: param_loop: Fixture which provides one set of values required to run the utility
         """
-        source_file = TEMP_PATH + 'source_file'
-        remote_file = TEMP_PATH + M0CRATE_WORKLOAD_YML.split("/")[-1]
+        source_file = TEMP_PATH + '/source_file'
+        remote_file = TEMP_PATH + '/' + M0CRATE_WORKLOAD_YML.split("/")[-1]
```
Comment (seagate-sarang-sawant) [inline comptests/motr/test_motr_k8s_sanity.py:86]: Use tempfile module and form the path on the fly in test.

File: libs/motr/motr_core_k8s_lib.py [hunk 3]
```diff
@@ -142,13 +142,23 @@ def get_primary_cortx_node(self):
         """
         motr_client_pod = self.node_obj.get_pod_name(
             pod_prefix=common_const.CLIENT_POD_NAME_PREFIX)[1]
-        cmd = " | awk -F ' '  '/(RC)/ { print $1 }'"
+        cmd1 = " | awk -F ' '  '/(RC)/ { print $1 }'"
```
Comment (seagate-sarang-sawant) [inline libs/motr/motr_core_k8s_lib.py:145]: Well cmd1, cmd2 and other names can replaced with more meaningful names.

File: libs/motr/motr_core_k8s_lib.py [hunk 4]
```diff
@@ -142,13 +142,23 @@ def get_primary_cortx_node(self):
...
         primary_cortx_node = self.node_obj.send_k8s_cmd(
             operation="exec", pod=motr_client_pod, namespace=common_const.NAMESPACE,
             command_suffix=f"-c {common_const.HAX_CONTAINER_NAME} "
-                           f"-- {common_cmd.MOTR_STATUS_CMD} {cmd}",
+                           f"-- {common_cmd.MOTR_STATUS_CMD} {cmd1}",
             decode=True)
-        return primary_cortx_node.replace("data", "client")
+        data_pod_name = primary_cortx_node.split('.')[0]
```
Comment (seagate-sarang-sawant) [inline libs/motr/motr_core_k8s_lib.py:151]: Abstract out k8s command complexity in another sub function.

**Review by seagate-sarang-sawant — CHANGES_REQUESTED** [review]

File: libs/motr/motr_core_k8s_lib.py (hunk 4 above)
Comment (zohebkhann) [reply libs/motr/motr_core_k8s_lib.py:151]: I think it is already available with **send_k8s_cmd** function

**Review by seagate-sarang-sawant — APPROVED** [review]
I am not able to see my prior comments.

**Comment by zohebkhann** [comment]
> I am not able to see my prior comments.

These are marked as resolved and now outdated

File: libs/motr/motr_core_k8s_lib.py (hunk 4 above)
Comment (seagate-sarang-sawant) [reply libs/motr/motr_core_k8s_lib.py:151]: the comments are not resolved .

Comment (seagate-sarang-sawant) [reply libs/motr/motr_core_k8s_lib.py:151]: I want you to colocate all k8s functions in a file for motr/ probably use builder pattern to construct commands. Will help in long term.

Comment (zohebkhann) [reply libs/motr/motr_core_k8s_lib.py:151]: As discussed, I will create another ticket for making these changes.

**Pull request merged**